cli/command_snapshot_create.go

```diff
@@ -91,17 +93,18 @@ func (c *commandSnapshotCreate) run(ctx context.Context, rep repo.RepositoryWrit
 		return errors.Wrap(err, "error upgrading repository")
 	}
 
-	if c.snapshotCreateAll {
+	switch {
+	case len(sources) == 0 && c.snapshotCreateAll:
```

Comment:
I can get rid of these new validations, just seemed like we have a few combinations of parameters that result in very unintuitive executions